core/domain/topic_domain.py

```diff
@@ -221,14 +236,107 @@ def validate(self):
                     % (skill_id, self.id))
 
 
+class SubtopicPage(object):
+    """Domain object for a Subtopic page."""
+
+    def __init__(self, subtopic_page_id, topic_id, html_data, language_code):
+        """Constructs a SubtopicPage domain object.
+
+        Args:
+            subtopic_page_id: str. The unique ID of the subtopic page.
+            topic_id: str. The topic that this subtopic is a part of.
+            html_data: str. The html data that describes this subtopic.
+            language_code: str. The ISO 639-1 code for the language this
+                sub topic page is written in.
+        """
+        self.id = subtopic_page_id
+        self.topic_id = topic_id
+        self.html_data = html_cleaner.clean(html_data)
+        self.language_code = language_code
+
+    def to_dict(self):
+        """Returns a dict representing this SubtopicPage domain object.
+
+        Returns:
+            A dict, mapping all fields of SubtopicPage instance.
+        """
+        return {
+            'id': self.id,
+            'topic_id': self.topic_id,
+            'html_data': self.html_data,
+            'language_code': self.language_code
+        }
+
+    @classmethod
+    def get_subtopic_page_id(cls, topic_id, subtopic_id):
+        """Returns the subtopic page id from the topic is and subtopic_id.
+
+        Args:
+            topic_id: str. The id of the topic that the subtopic is a part of.
+            subtopic_id: int. The id of the subtopic.
+
+        Returns:
+            str. The subtopic_page_id calculated from the given values.
+        """
+        return '%s-%s' % (topic_id, subtopic_id)
+
+    @classmethod
+    def create_default_subtopic_page(cls, subtopic_page_id, topic_id):
+        """Creates a SubtopicPage object with default values.
+
+        Args:
+            subtopic_page_id: str. ID of the new subtopic page.
+            topic_id: str. The topic to which this page is linked with.
+
+        Returns:
+            SubtopicPage. A subtopic object with given id, topic_id and empty
+                html_data field.
+        """
+        return cls(
+            subtopic_page_id, topic_id, '', constants.DEFAULT_LANGUAGE_CODE)
+
+    def update_html_data(self, new_html_data):
+        """The new value for the html data field.
+
+        Args:
+            new_html_data: str. The new html data for the subtopic page.
+        """
+        self.html_data = new_html_data
```

Comment:
Just to check -- suppose we create a new subtopic page and add some HTML into it. Do we clean it before saving?

Reply:
No, added the cleaning.